nipype/pipeline/engine/utils.py

```diff
@@ -1485,7 +1485,7 @@ def clean_working_directory(
             if f not in needed_files:
                 if not needed_dirs:
```

Comment:
Since `needed_dirs` is a list and `"x".startswith(()) == False`:

```suggestion
                if not f.startswith(tuple(needed_dirs)):
```